solr/core/src/java/org/apache/solr/update/processor/DistributedZkUpdateProcessor.java

```diff
@@ -114,16 +104,29 @@
   private RollupRequestReplicationTracker rollupReplicationTracker;
   private LeaderRequestReplicationTracker leaderReplicationTracker;
 
+  /**
+   * For {@link Replica.Type#SHARED} replica, it is necessary that we pull from the shared store at the start of
+   * an indexing batch (if the core is stale). And we push to the shared store at the end of a successfully committed
+   * indexing batch (we ensure that each batch has a hard commit). Details can be found in 
+   * {@link org.apache.solr.store.shared.SharedCoreConcurrencyController}.
+   * In other words, we would like to call {@link SharedCoreIndexingBatchProcessor#startIndexingBatch()} at the start of
```

Comment:
Also startIndexingBatch and finishIndexingBatch are wrapped around more public methods addOrDeleteGoingToBeIndexedLocally or hardCommitCompletedLocally that are actually used in this class. Should we reference those methods in the doc instead?

Reply:
I never liked this place to run batch start/finish logic. I have added a TODO.

The doc also refers addOrDeleteGoingToBeIndexedLocally and hardCommitCompletedLocally few lines later.
startIndexingBatch and finishIndexingBatch are mentioned first because they are the real reason for whole logic. They are not public because they don't need to be. But if/when we find another better place to start/finish a batch we will likely make them public and directly call them.